docs: add tags column to ai integrations (#4509)
temporalio/documentation: docs/develop/python/integrations/index.mdx

Reviewer: ```suggestion
title: AI Integrations
```

Reviewer: ```suggestion
sidebar_label: AI Integrations
```

Reviewer 2: While we're here--it has been bothering me that the sidebar says "Integrations" but the title is "AI Integrations".
Now that we're adding these tags and using the word AI there ("AI Framework"), maybe it's okay to just use "Integrations."  This will also allow us to add the popular OTel plugin.
Of course, we can continue to use AI integrations in the keywords for this page for SEO.

Reviewer: I agree with this proposal

Author: ```suggestion
title: Integrations
```

Author: ```suggestion
sidebar_label: Integrations
```

---
id: index

> title: AI Integrations

Author: ```suggestion
title: AI integrations
```

> sidebar_label: AI Integrations

Author: ```suggestion
sidebar_label: AI integrations
```

> The following AI framework integrations are available for the Temporal Python SDK:

Reviewer: ```suggestion
The following integrations are available between the Temporal Python SDK and third-party AI frameworks and tooling:
```

Reviewer: (not every thing listed is an AI framework)
